src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,119 @@
+        counter= counter + 1;
+        System.out.println("Got it. I've added this task:\n" + helper(action, "T", false));
+        System.out.println("Now you have " + counter + " tasks in the list.");
+    }
+
+
+    /**
+     * Helper method to format the type of action.
+     * @param task Name of task.
+     * @param taskType Type of task.
+     * @param isDone True if done, false if not done.
+     * @return String with formatted type of tasktype.
+     */
+    public static String helper(String task, String taskType, boolean isDone) {
+        String taskIcon;
+        if (taskType.equals("T")) {
+            taskIcon = "[T]";
+        } else if (taskType.equals("D")) {
+            taskIcon = "[D]";
+        } else if (taskType.equals("E")) {
+            taskIcon = "[E]";
+        } else {
+            taskIcon = "[ ]";
+        }
+        return taskIcon + "[" + (isDone ? "X" : " ") + "] " + task;
+    }
+
+    /**
+     * Helper method to format and save the deadline type of task.
+     * @param action Name of task.
+     * @param due String of when the task is due.
+     * @param by LocalDateTime of formatted due.
+     */
+    public static void deadline(String action, String due, LocalDateTime by) {
+        String byString = due.trim();
+        dueString[counter] = byString;
+        actions[counter] = action;
+        // System.out.println(action + "hello");
+        type[counter] = "D";
+        isDone[counter] = false;
+        counter = counter + 1;
+        System.out.println("Got it. I've added this task: \n" + helper(action, "D", false) + " (by" + " " + byString + ")");
+        System.out.println("Now you have " + counter + " tasks in the list.");
+
+    }
+
+    /**
+     * Helper to format and save action of event type.
+     * @param action Name of action.
+     * @param start String indicating start time.
+     * @param end String indicating the end time.
+     */
+    public static void event(String action, String start, String end) {
+        String from = start.trim();
+        String to = end.trim();
+        LocalDateTime startTimeEvent = helper2(from);
+        LocalDateTime endTimeEvent = helper2(to);
+        startTime[counter] = startTimeEvent;
+        endTime[counter] = endTimeEvent;
+        actions[counter] = action;
+        isDone[counter] = false;
+        type[counter] = "E";
+        counter = counter + 1;
+        System.out.println("Got it. I've added this task:\n" + helper(action, "E", false) + " (from: " + from + " to: " + to + ")");
+        System.out.println("Now you have " + counter + " tasks in the list.");
+    }
+
+    /**
+     * Helper method to format the string into the LocalDateTime format.
+     * @param dateTimeStr String of the to-be-formatted date time.
+     * @return Formatted LocalDateTime version.
+     */
+    private static LocalDateTime helper2 (String dateTimeStr) {
+        DateTimeFormatter DMYhelper = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```

Comment:
Shouldn't this be in camelCase?

```suggestion
        DateTimeFormatter dmyHelper = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```